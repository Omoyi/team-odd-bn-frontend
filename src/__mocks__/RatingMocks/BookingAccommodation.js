export const RateBookingsMocks = {
  popUpAction: jest.fn(),
  display: {
    rate: 'flex',
  },
  initialPopupState: {
    popUpDisplay: {
      rate: 'flex',
    },
  },
  state: {
    popUpDisplay: {
      rate: 'none',
    },
  },
  onEditInputAction: jest.fn(),
};

export const viewSpecificAccommodationMocks = {
  singleAccommodation: jest.fn(),
  match: {
    params: {
      id: 1,
    },
  },
  accommodationItem: {
    id: 45,
    userId: 15,
    name: 'Eric Palace',
    cityId: 2,
    address: 'Kigali',
    description: 'I am going there',
    googleCoordinates: '-22,-12',
    createdAt: '2020-01-24T14:29:50.618Z',
    updatedAt: '2020-01-24T14:29:50.618Z',
  },
  accommodationItemError: [],
};

export const noSpecificAccommodationMocks = {
  singleAccommodation: jest.fn(),
  match: {
    params: {
      id: 1,
    },
  },
  accommodationItem: undefined,
  accommodationItemError: ['server error'],
};

export const accommodationReviewsMocks = {
  accommodationItem: {
    ratings: [
      {
        id: 39,
        userId: 1,
        accommodationId: 45,
        rating: '3',
        review: 'Not really bad ',
        createdAt: '2020-02-04T06:10:04.753Z',
        updatedAt: '2020-02-04T06:10:04.753Z',
        users: {
          id: 1,
          firstName: 'John',
          lastName: 'Doe',
          userProfile: {
            imageURL: 'https://res.cloudinary.com/dpzertkmu/image/upload/v1579680509/clientImage/kristopher-roller-PC_lbSSxCZE-unsplash_mcyct9.jpg',
          },
        },
      },
    ],
  },
};

export const emptyAccommodationReviewsMocks = {
  accommodationItem: {
    ratings: [],
  },
};

export const mapStateToProspReviews = {
  state: {
    specificAccommodation: {
      singleAccommodation: [
        { rates: 'accommodation items follows' },
      ],
    },
  },
};

export const accommodationRoomsProps = {
  accommodationItem: {
    id: 1,
    accommodationRooms: [
      {
        id: 88,
        accommodationId: 45,
        name: 'Kackity',
        roomType: 'Double',
        createdAt: '2020-01-24T14:29:50.761Z',
        updatedAt: '2020-01-24T14:29:50.761Z',
      },
      {
        id: 89,
        accommodationId: 45,
        name: 'Serena',
        roomType: 'Single',
        createdAt: '2020-01-24T14:29:50.762Z',
        updatedAt: '2020-01-24T14:29:50.762Z',
      },
    ],
    imagesAccommodation: [
      {
        id: 61,
        accommodationId: 45,
        imageUrl: 'https://images.unsplash.com/photo-1559599238-308793637427?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
        createdAt: '2020-01-24T14:29:50.657Z',
        updatedAt: '2020-01-24T14:29:50.657Z',
      },
      {
        id: 62,
        accommodationId: 45,
        imageUrl: 'https://images.unsplash.com/photo-1558685385-e9ad7b3c885c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
        createdAt: '2020-01-24T14:29:50.665Z',
        updatedAt: '2020-01-24T14:29:50.665Z',
      },
    ],
  },
};

export const roomsImagesPrevProps = {
  prevProps: {
    accommodationItem: {
      id: 2,
      accommodationRooms: [
        {
          id: 88,
          accommodationId: 45,
          name: 'Kackity',
          roomType: 'Double',
          createdAt: '2020-01-24T14:29:50.761Z',
          updatedAt: '2020-01-24T14:29:50.761Z',
        },
      ],
      imagesAccommodation: [
        {
          id: 61,
          accommodationId: 45,
          imageUrl: 'https://images.unsplash.com/photo-1559599238-308793637427?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
          createdAt: '2020-01-24T14:29:50.657Z',
          updatedAt: '2020-01-24T14:29:50.657Z',
        },
      ],
    },
  },
};

export const viewSpecificAccommodationViewsMocks = {
  accommodationItem: {
    id: 45,
    userId: 15,
    name: 'Eric Palace',
    cityId: 2,
    address: 'Kigali',
    description: 'I am going there',
    googleCoordinates: '-22,-12',
    createdAt: '2020-01-24T14:29:50.618Z',
    updatedAt: '2020-01-24T14:29:50.618Z',
    accommodationRooms: [
      {
        id: 88,
        accommodationId: 45,
        name: 'Kackity',
        roomType: 'Double',
        createdAt: '2020-01-24T14:29:50.761Z',
        updatedAt: '2020-01-24T14:29:50.761Z',
      },
      {
        id: 89,
        accommodationId: 45,
        name: 'Serena',
        roomType: 'Single',
        createdAt: '2020-01-24T14:29:50.762Z',
        updatedAt: '2020-01-24T14:29:50.762Z',
      },
    ],
    city: {
      id: 2,
      city: 'Kigali',
      createdAt: '2019-12-09T07:36:36.417Z',
      updatedAt: '2019-12-09T07:36:36.417Z',
    },
    imagesAccommodation: [
      {
        id: 61,
        accommodationId: 45,
        imageUrl: 'https://images.unsplash.com/photo-1559599238-308793637427?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
        createdAt: '2020-01-24T14:29:50.657Z',
        updatedAt: '2020-01-24T14:29:50.657Z',
      },
      {
        id: 62,
        accommodationId: 45,
        imageUrl: 'https://images.unsplash.com/photo-1558685385-e9ad7b3c885c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
        createdAt: '2020-01-24T14:29:50.665Z',
        updatedAt: '2020-01-24T14:29:50.665Z',
      },
    ],
  },
  popUpAction: jest.fn(),
  display: {
    map: 'flex',
  },
};
