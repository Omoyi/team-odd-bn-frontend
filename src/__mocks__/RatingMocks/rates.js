export const rateData = {
  onEditInputAction: jest.fn(),
  popUpAction: jest.fn(),
  ratings: {
    starRateInput: 0,
    feedbackInput: '',
    accommodationId: '',
    ratingError: null,
    submitSucces: false,
  },
  submitRatingAction: jest.fn(),
};

export const mapStateToPropsMockData = {
  ratings: {
    RatesAccommodation: {
      starRateInput: 0,
      feedbackInput: '',
      accommodationId: '',
      ratingError: null,
      submitSucces: false,
    },
  },
};

export const mockActionData = {
  inputDataMock: {
    type: 'any_action',
    payload: 'any_data',
  },
  initialState: {
    RatesAccommodation: {
      starRateInput: 0,
      feedbackInput: '',
      accommodationId: '',
      ratingError: null,
      submitSucces: false,
    },
  },
  addNewRateMessage: {
    RatesAccommodation: {
      starRateInput: 0,
      feedbackInput: 'I like the accommodation',
      accommodationId: '',
      ratingError: null,
      submitSucces: false,
    },
  },
  submitRateSuccessMock: [
    {
      payload: 'I am data',
      type: 'ADD_ACCOMMODATION_RATE_INPUT',
    },
    {
      payload: { submitSucces: true },
      type: 'RATE_ACCOMMODATION_SUCCESS',
    },
  ],
  successfullySendRateFeedback: {
    RatesAccommodation: {
      starRateInput: 0,
      feedbackInput: '',
      accommodationId: '',
      ratingError: null,
      submitSucces: true,
    },
  },
  errorRejectSendRateFeedback: {
    RatesAccommodation: {
      starRateInput: 0,
      feedbackInput: '',
      accommodationId: '',
      ratingError: 'Your have submit feedback already',
      submitSucces: false,
    },
  },
};
