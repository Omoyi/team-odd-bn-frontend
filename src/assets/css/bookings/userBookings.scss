.userBookingContainer{
  min-height: 100%;
  width: 100%;
}
.userBookingsHeader {
  display: flex;
  flex-direction: row;
  @media screen and (min-width: 200px) and (max-width: 750px) {
    flex-direction: column;
  }
  font-size: 30px;
  letter-spacing: 0.5px;
  text-align: center;
  font-weight: bold;
  font-family: muli;
  height: 130px;
  width: 80%;
  margin: auto;
  padding: 0px 0 1rem 0px;
  justify-content: center;
  div {
    height: 40%;
    margin: auto 0px;
    input {
      height: 43px;
      width: 191px;
      width: 100%;
      margin: auto;
    }
  }
  .bookingTitle {
    font-weight: bold;;
  }
  .bookingSearch {
  }
}
.userBookingsContainer {
  display: flex;
  flex-direction: column;
  min-height: 64%;
  width: 80%;
  margin: auto;
  
  .bookingHeader{
    font-size: 24px;
    margin-right: -4%;
    margin: 5px 0;
    padding: 0;
    font-family: muli;
    font-weight: bold;
    letter-spacing: 0.8px;
  }
  .bookingDetails{
    letter-spacing: 0.5px;
    font-weight: bold;
   
    .rateBookingsBtn {
      margin-top: -100px;
      color: #7D7D7D;
      font-weight: 900;
      padding-top: -10px;
    }
    .rateBookingsBtn button {
      float: right;
      background-color: #32c8f4;
      font-size: 1.1rem;
      color: #fff;
      border-radius: 7px;
      height: 40px;
      box-shadow: 1px 2px 3px 0px #BBBBBB;
      cursor: pointer;
      margin-top: 40px;
      width: 130px;
    }
    .rateBookingsBtn button:hover {
      background-color: #fff;
      color: #32c8f4;
      border: 1px solid #32c8f4;
    }
      .booking-like-contents {
        margin-top: 0px ;
        .like-dislikes{
          width: 25%;
          .like-container{
            img{
              width: 25px;
              height: 25px;
            }
            img:hover{
              cursor: pointer;
            }
          }
          .dislike-container{
            img{
              width: 25px;
              height: 25px;
              margin-left: -16px;
            }
            img:hover{
              cursor: pointer;
            }
          }
        }
        .likes_dislikes_data{
          display: flex;
          width: 13%;
          justify-content: space-between;
        }
        .rateBookingsBtn button{
          margin-top: -10px;
          margin-right: 30px;
        }
        
      }
    img{
      padding-right: 8px;
    }
    padding-left: 20px;

    .cityBooking, .roomBookings, .checkOutBookings, .checkInBookings{
      display: flex;
      letter-spacing: 0.5px;
      font-weight: bold;
      font-family: Muli;
      font-style: normal;
      font-weight: bold;
      font-size: 15px;

      img{
        font-weight: bold;
        width: 19px;
        height: 20px;
      }
  
    }
    .createdOnBooking p{
      color: #828282;
      font-style: normal;
      margin-right: 50px;
      float: right;
      font-size: 12px;
    }
    
  }
  .singleBooking {
    display: flex;
    flex-direction: row;
    height: 260px;
    @media screen and (min-width: 200px) and (max-width: 750px) {
      flex-direction: column;
      height: auto;
    }
    width: 99%;
    margin: 10px auto;
    box-shadow: 1px 2px 2px rgba(0, 0, 0, 0.25);
    .bookingImage {
      width: 45%;
     
      @media screen and (min-width: 200px) and (max-width: 750px) {
        width: 100%;
      }
      img {
        width: 100%;
        height: 100%;
      }
    }
    .bookingImage :hover {
      cursor: pointer;
    }
    .bookingImage img {
      border: 0.2px solid #ddd;
      border-radius: 2px;
    }
    .bookingImage img:hover {
      box-shadow: 0 0 2px 1px rgba(0, 140, 186, 0.5);
    }
    .bookingDetails {
      width: 70%;
      @media screen and (min-width: 200px) and (max-width: 750px) {
        width: 100%;
      }
      height: 100%;
      display: flex;
      flex-direction: column;
      div {
        width: 100%;
        height: 100%;
      }
    }
  }

  .userBookingsFooter {
    width: 80%;
    .pagination {
      display: flex;
      flex-direction: row;
      justify-content: center;
      .pageArrows {
        max-width: 40px;
        button {
          color: #00b9f2;
          background-color: white;
          border-radius: 50%;
          outline: none;
          width: 35px;
          height: 35px;
          padding: 1px;
          margin: 2px 1px;
          a {
            display: block;
            width: 100%;
            height: 100%;
            text-decoration: none;
            color: #00b9f2;
            padding: 5px 1px;

            border-radius: 50%;
            outline: none;
            width: 33px;
            height: 33px;
         
          }
        }
      }
      .pages {
        overflow-x: auto;
        max-width: 60%;
        button {
          background-color: white;
          color: #00b9f2;
          border-radius: 50%;
          outline: none;
          width: 35px;
          height: 35px;
          padding: 1px;
          margin: 1px;
          a {
            display: block;
            width: 100%;
            height: 100%;
            text-decoration: none;
            color: #00b9f2;
            padding: 5px 0px;
          }
        }
      }
    }
  }
}

.empty-bookings-container {
  width: 50%;
  margin: auto;
  padding: 10px 0 10px 5px;
  border: green solid 1px !important;
  border-radius: 4px;
  
  .error-message-booking {
    font-size: 17px;
    margin: 20px 0px 10px 0;
    text-align: center;
  }
}

@media screen and (min-width: 360px) and (max-width: 575px) {
  .empty-bookings-container {
    width: 90%;
    padding: 10px 0 10px 5px;
  }
}

@media screen and (min-width: 576px) and (max-width:760px) {
  .empty-bookings-container {
    width: 80%;
  }
}
@media screen and (min-width: 760px) and (max-width: 900px) {
  .empty-bookings-container {
    width: 60%;
  }
}
@media (min-width: 900px) {
  .empty-bookings-container {
    width: 50%;
  }
}
