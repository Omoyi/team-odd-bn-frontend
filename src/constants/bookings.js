import React from 'react';

export default (
  availlableRoomTypes,
  accommodationId,
  availlableRooms,
  roomId,
  updateBookAccommodationInput,
  roomType,
) => [
  {
    label: 'Room type',
    inputField:
  <select
    key="selectRoomType"
    id="selectRoomType"
    onChange={(e) => updateBookAccommodationInput({ roomType: e.target.value })}
    disabled={!accommodationId}
  >
    <option>Select room type</option>
    {availlableRoomTypes
       && availlableRoomTypes.map((room) => (
         <option value={room} key={room}>
           {room}
         </option>
       ))}
  </select>,
  },
  {
    label: 'Room name',
    inputField:
  <select
    key="selectRoomName"
    id="selectRoomName"
    onChange={(e) => updateBookAccommodationInput({ roomId: parseInt(e.target.value, 10) })}
    disabled={!roomType}
  >
    <option>Select room name</option>
    {availlableRooms
       && availlableRooms.map((room) => (
         <option value={room.id} key={room.name}>
           {room.name}
         </option>
       ))}
  </select>,
  },
  {
    label: 'Check-in date',
    inputField: <input
      key="checkInDate"
      id="checkInDate"
      onChange={(e) => updateBookAccommodationInput({ checkInDate: e.target.value })}
      disabled={!roomId}
      type="date"
    />,
  },
  {
    label: 'Check-out date',
    inputField: <input
      key="checkOutDate"
      id="checkOutDate"
      onChange={(e) => updateBookAccommodationInput({ checkOutDate: e.target.value })}
      disabled={!roomId}
      type="date"
    />,
  },
];
