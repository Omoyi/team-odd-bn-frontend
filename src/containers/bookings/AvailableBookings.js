import React, { Component } from 'react';
import { connect } from 'react-redux';
import Dashboard from '../../views/Dashboard/sidebar/index';
import RatesBookings from '../../views/bookings/RatesBookings';
import popUpAction from '../../redux/actions/popUpAction';
import PopUp from '../../components/PopUp';
import { onEditInputAction } from '../../redux/actions/ratingAction';

export class AvailableBookings extends Component {
  render() {
    const { popUpAction, display, onEditInputAction } = this.props;
    return (
      <Dashboard>
        <button
          type="button"
          className="form-btn-open"
          onClick={() => {
            popUpAction({
              currentPopUp: 'rate',
              rate: 'flex',
            });
            onEditInputAction({ accommodationId: Math.random(10, 100) });
          }}
        >
click me
        </button>
        <PopUp popUp={<RatesBookings />} display={display.rate} />
      </Dashboard>
    );
  }
}

export const mapStateToProps = (state) => ({
  display: state.popUpsDisplay,
});

export default connect(mapStateToProps, { popUpAction, onEditInputAction })(AvailableBookings);
