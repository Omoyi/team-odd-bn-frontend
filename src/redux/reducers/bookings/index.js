import { combineReducers } from 'redux';
import getAllBookingReducers from './getAllBookingReducers';

export default combineReducers({
  bookings: getAllBookingReducers,
});
