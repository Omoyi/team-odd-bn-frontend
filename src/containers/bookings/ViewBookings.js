import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Dashboard from '../../views/Dashboard/sidebar';
import { getAllBookings } from '../../redux/actions/bookings/getAllBookingsActions';
import { Paginate } from '../../helpers/Paginate';
import { changePageNo } from '../../redux/actions/PaginationAction';
import '../../assets/css/bookings/userBookings.scss';
import icon1 from '../../assets/images/ViewAllBookings/icon1.png';
import icon2 from '../../assets/images/ViewAllBookings/icon2.png';
import icon3 from '../../assets/images/ViewAllBookings/icon3.png';
import icon4 from '../../assets/images/ViewAllBookings/icon4.png';
import { likeAction, dislikeAction } from '../../redux/actions/likeAndDislikeAction';
import { viewActionAccommodation } from '../../redux/actions/allAccommodation';
import likeIcon from '../../assets/images/like_icon/thumbs-up_32.png';
import disLikeIcon from '../../assets/images/dislike_icon/thumbs-down_32.png';
import joinLikesAccomodation from '../../helpers/joinLikesAccomodation'
import { getCitiesAction } from '../../redux/actions/createAccommodationActions';
import { openSingleBooking, getSingleBooking } from '../../redux/actions/bookings/viewSingleBookingActions';
import ViewSingleBooking from '../../containers/bookings/viewSingleBooking';
import { onEditInputAction } from '../../redux/actions/ratingAction'
import popUpAction from '../../redux/actions/popUpAction';
import PopUp from '../../components/PopUp';
import RatesBookings from '../../views/bookings/RatesBookings';
import Pagination from 'custom_react_pages'

export class ViewBookings extends Component {

  state = {
    modal: false,
    index: null,
  }
  async componentDidMount() {    
    this.props.getCitiesAction();
    const { viewActionAccommodation } = this.props;
    viewActionAccommodation();
    await this.props.getAllBookings();
    const { allAccomodation } = this.props.accommodation;
    this.setState({ allAccomodation });
  }
  
  handleSubmit  = (e, id, index) => {  
    e.preventDefault();
    this.setState(prevState=>(
      {
        modal: !prevState.modal,
        index
      }
    ))
    this.props.openSingleBooking(true);
    this.props.getSingleBooking(id.id);
  }

  handleClick = async(like, item) =>{       
    const { likeAction,viewActionAccommodation, dislikeAction } = this.props;

    if(like === "true"){
    const passData =  await likeAction({like: true, id: item})
      const updateView = await viewActionAccommodation()
      return  { passData, updateView };
    }
    else if(like === "false"){
      const passData = await dislikeAction( {dislike: true, id: item})
      const updateView =  await viewActionAccommodation();
      return  { passData, updateView };
    }
    return await viewActionAccommodation();
  }

  render() {
  
    const {
      allBookings, pageNo, itemsPerPage, changePageNo, allAccomodation, popUpAction, display, onEditInputAction
    } = this.props;
    const { accommodations, allLikes } = allAccomodation;

    allLikes.map((like, index) => Object.assign(accommodations[index], like));
    
    const accommodationsData = allBookings ? allBookings.data.myBookingsresult.map(item=>item):[];
    
    joinLikesAccomodation(accommodations,allLikes).map((item)=>{
      accommodationsData.forEach(element => {
        if(item.id === element.room.accommodationId){
          Object.assign(element,item)
        }
      });
    })
  
    const chunkBookings = Paginate(accommodationsData, itemsPerPage);
    const bookingsPages = { ...chunkBookings };
    const currentPage = bookingsPages[pageNo];
    

    return (
      <Dashboard>
        <div className="userBookingContainer">
          <div className="userBookingsHeader">
            <div className="bookingTitle">Bookings</div>
          </div>
          {this.props.allBookings ? (
            <div className="userBookingsContainer">
              {this.state.modal && (
                <ViewSingleBooking
                  handleUpdateModal={item => this.setState({ modal: !item })}
                  myIndex={this.state.index}
                  modal={this.state.modal}
                />
              )}

              {accommodationsData.length > 0 && (
                <Pagination
                  itemsPerPage={6}
                  next="next"
                  prev="previous"
                  data={accommodationsData.length ? accommodationsData : []}
                  oneItem={(booking, index) => (
                    <div key={index} className="singleBooking">
                      <div
                        className="bookingImage"
                        onClick={e => {
                          this.handleSubmit(e, booking, index)
                        }}
                      >
                        <img
                          src={
                            booking.room.accommodation.imagesAccommodation[0]
                              .imageUrl
                          }
                        />
                      </div>
                      <div className="bookingDetails">
                        <div className="bookingHeader">
                          {booking.room.accommodation.name}
                        </div>
                        <div className="checkInBookings">
                          <img src={icon2} alt="icon1" />
                          <p>
                            Check-in date:
                            {booking.checkInDate.slice(0, 10)}
                          </p>
                        </div>
                        <div className="checkOutBookings">
                          <img src={icon1} alt="icon2" />
                          <p>
                            Check-out date:
                            {booking.checkOutDate.slice(0, 10)}
                          </p>
                        </div>
                        <div className="roomBookings">
                          <img src={icon3} alt="icon3" />
                          <p>
                            {`${booking.room.roomType} | ${booking.room.name}`}
                          </p>
                        </div>
                        <div className="cityBooking">
                          <img src={icon4} alt="icon4" />
                          <p>{booking.room.accommodation.city.city}</p>
                        </div>

                        <div className="booking-like-contents">
                          <section className="common-content like-dislikes">
                            <div className="like-dislike-contents like-container">
                              <img
                                onClick={() => {
                                  const item = booking.room.accommodationId

                                  return this.handleClick('true', item)
                                }}
                                className="likes_icon_img"
                                src={likeIcon}
                                alt="like icon"
                              />
                            </div>
                            <div className="like-dislike-contents dislike-container">
                              <img
                                onClick={() => {
                                  const item = booking.room.accommodationId
                                  this.handleClick('false', item)
                                }}
                                className="dislike_icon"
                                src={disLikeIcon}
                                alt="dislike icon"
                              />
                            </div>
                          </section>
                          <section className="common-content likes_dislikes_data">
                            <span className="likes_side">
                              {booking.likeCounter}
                            </span>
                            <span className="dislikes_side">
                              {booking.dislikeCounter}
                            </span>
                          </section>

                          <div className="rateBookingsBtn">
                            <button
                              className="rateBookingBtn"
                              type="button"
                              className="form-btn-open"
                              onClick={() => {
                                popUpAction({
                                  currentPopUp: 'rate',
                                  rate: 'flex',
                                })
                                onEditInputAction({
                                  accommodationId: booking.id,
                                })
                              }}
                            >
                              Rate
                            </button>
                          </div>
                        </div>

                        <div className="createdOnBooking">
                          <p>
                            Request Date:
                            {booking.createdAt.slice(0, 10)}
                          </p>
                        </div>
                      </div>
                    </div>
                  )}
                />
              )}
            </div>
          ) : (
            <div className="empty-bookings-container">
              <p className="error-message-booking">
                You do not have bookings click
                <Link to="/accommodations"> here </Link>
                to book new one
              </p>
            </div>
          )}

          <PopUp popUp={<RatesBookings />} display={display.rate} />
        </div>
      </Dashboard>
    )
  }
}
export const mapStateToProps = (state, { userLikesAndDislike }) => ({
  allBookings: state.bookings.bookings.allBookings,
  pageNo: state.pagination.pageNo,
  itemsPerPage: state.pagination.itemsPerPage,
  likeDislikeState: userLikesAndDislike,
  allAccomodation: state.allAccomodation,
  bookings: state.bookings,
  display: state.popUpsDisplay,
});

export default connect(mapStateToProps, {
  getAllBookings, changePageNo, viewActionAccommodation, likeAction, 
  dislikeAction, openSingleBooking, getSingleBooking, getCitiesAction, 
  popUpAction, onEditInputAction
})(
  ViewBookings,
);
