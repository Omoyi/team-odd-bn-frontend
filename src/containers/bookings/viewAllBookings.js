import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getCitiesAction } from '../../redux/actions/createAccommodationActions';
import { openSingleBooking, getSingleBooking } from '../../redux/actions/bookings/viewSingleBookingActions';
import ViewSingleBooking from './viewSingleBooking';
import Dashboard from '../../views/Dashboard/sidebar/index';

export class AllBookings extends Component {
  componentDidMount() {
    this.props.getCitiesAction();
  }

  handleSubmit(e, id) {
    e.preventDefault();
    this.props.openSingleBooking(true);
    this.props.getSingleBooking(id);
  }


  render() {
    return (
      <Dashboard>
        <div className="tempStyle">
          {this.props.bookings.modal.openSingleBooking && <ViewSingleBooking />}
          <button
            data-test="view-single-booking"
            type="button"
            onClick={(e) => {
              this.handleSubmit(e, 1);
            }}
          >
            View
          </button>
        </div>
      </Dashboard>
    );
  }
}
export const mapStateToProps = (state) => ({
  bookings: state.bookings,
});
export default connect(mapStateToProps, { openSingleBooking, getSingleBooking, getCitiesAction })(AllBookings);
