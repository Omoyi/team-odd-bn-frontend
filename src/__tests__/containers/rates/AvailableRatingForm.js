/* eslint-disable import/no-duplicates */
import React from 'react';
import { shallow } from 'enzyme';
import { AvailableBookings, mapStateToProps } from '../../../containers/bookings/AvailableBookings';
import { RateBookingsMocks } from '../../../__mocks__/RatingMocks/BookingAccommodation';

describe('Expect to popup the rating form', () => {
  const wrapper = shallow(<AvailableBookings {...RateBookingsMocks} />);
  it('Rate form should render successfully', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Rate form should open', () => {
    const findBtn = wrapper.find('.form-btn-open');
    findBtn.simulate('click');
    expect(findBtn).toBeTruthy();
  });

  it('Should mock booking form map state to props', () => {
    const result = mapStateToProps(RateBookingsMocks.state);
    expect(result).toEqual({ display: undefined });
  });
});
