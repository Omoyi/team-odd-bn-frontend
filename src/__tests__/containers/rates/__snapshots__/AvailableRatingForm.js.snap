// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to popup the rating form Rate form should render successfully 1`] = `ShallowWrapper {}`;
